internal/app/machined/pkg/controllers/k8s/render_config_static_pods.go

```diff
@@ -93,11 +93,15 @@ func (ctrl *RenderConfigsStaticPodController) Run(ctx context.Context, r control
 		for _, pod := range []struct {
 			name      string
 			directory string
+			uid       int
+			gid       int
 			configs   []configFile
 		}{
 			{
 				name:      "kube-apiserver",
 				directory: constants.KubernetesAPIServerConfigDir,
+				uid:       constants.KubernetesAPIServerRunUser,
```

Comment:
I guess this is for the user defined pods (I could be wrong here). If that's the case, we should not use defaults here

Reply:
These are configuration manifests (in this case only `admission-control-config.yaml`). Which must be to be readable by the APIServer pod.

As they are mounted in the pod:
```
  - hostPath:
      path: /system/config/kubernetes/kube-apiserver
      type: ""
    name: config
 ```
 
I have effectively only changed it from `constants.KubernetesRunUser` to `constants.KubernetesAPIServerRunUser` see: https://github.com/siderolabs/talos/pull/5917/files/a64941c7613289343e46385dc23c637190be2787#diff-c8111a0cf068d2938606c06f94680cc646d70d2aa4c2f32339aed1e409f076d3L131

Reply:
Thanks for proving me wrong, I thought this was for user defined pods :sweat_smile: